pkg/tfgen/generate.go

```diff
@@ -1786,3 +1797,18 @@ func ignoreMappingError(s []string, str string) bool {
 	}
 	return false
 }
+
+// Add a file to be emitted next to the schema.json only when the schema is itself
+// emitted.
+//
+// `panic`s if called with a `path` that conflicts with an existing file, including
+// `schema.json`.
+func AdditionalSchemaFile(path string, bytes []byte) {
```

Comment:
I've used https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pf/tfgen/main.go#L40-#L40 for this purpose before which does not feel quite right either.